packages/core/src/utilities/renderToCanvasGPU.ts

```diff
@@ -39,28 +39,29 @@ export default function renderToCanvasGPU(
   const viewportId = `renderGPUViewport-${imageIdToPrint}`;
   const imageId = image.imageId;
   const element = document.createElement('div');
-  element.style.width = `${canvas.width}px`;
-  element.style.height = `${canvas.height}px`;
+  const devicePixelRatio = window.devicePixelRatio || 1;
+  const originalWidth = canvas.width;
+  const originalHeight = canvas.height;
+  // Account for rounding errors by adding a half device pixel
```

Comment:
can you add more about what is rounding error and why half devide pixel ratio would help (what is the problem that it would even help)